### ISSUE TITLE
Make WeightsEnum public

### ISSUE THREAD

**Issue by adamjstewart** [pytorch/vision#7094]
### 🚀 The feature

I propose making `torchvision.models._api.WeightsEnum` public and adding documentation for it.

### Motivation, pitch

TorchGeo is adopting torchvision's new multi-weight support API: https://github.com/microsoft/torchgeo/pull/917

All of our weights subclass `WeightsEnum`, but this requires us to import directly from `torchvision.models._api`. Since this module is prefixed by an underscore, I'm assuming this class is private and subject to change. Also, since this class isn't documented, we have to silence Sphinx warnings about the missing parent class docs.

### Alternatives

We could copy-n-paste your `WeightsEnum` implementation but code duplication is bad and I don't want to deal with legal. We could keep importing from a private module but it may change in the future. Are there any other suggested alternatives for other libraries trying to copy the same design? Maybe this could be moved into TorchData?

### Additional context

@nilsleh

**Comment by NicolasHug** [comment]
Thanks for the feature request @adamjstewart . It's great to see torchgeo adopt the same API!

I think we're open to make the `Weights` and `WeightsEnum` classes public. They might require a bit of cleanup before we can make them public, so I started doing that in https://github.com/pytorch/vision/pull/7100.

**Comment by adamjstewart** [comment]
On the topic of WeightsEnum, I noticed that there is no md5 attribute and no checksumming done when downloading weights like there is for datasets. Should there be? (I can open a separate issue for this if you want)

**Comment by NicolasHug** [comment]
Sure @adamjstewart , that sounds reasonable. A new issue sounds like the best way forward.

**Comment by NicolasHug** [comment]
I believe this was addressed in https://github.com/pytorch/vision/pull/7100 and thus this issue can be closed
